src/zig.rs

```diff
@@ -166,6 +186,152 @@ impl zed::Extension for ZigExtension {
+            .and_then(|request| {
+                request.as_str().and_then(|s| match s {
+                    "launch" => Some(StartDebuggingRequestArgumentsRequest::Launch),
+                    "attach" => Some(StartDebuggingRequestArgumentsRequest::Attach),
+                    _ => None,
+                })
+            })
+            .ok_or_else(|| {
+                "Invalid request, expected 'request' to be either 'launch' or 'attach'".to_string()
+            })
+    }
+
+    fn dap_config_to_scenario(
+        &mut self,
+        zed_scenario: zed::DebugConfig,
+    ) -> Result<DebugScenario, String> {
+        match zed_scenario.request {
+            zed::DebugRequest::Launch(launch) => {
+                let config = ZigDebugConfig {
+                    program: Some(launch.program),
+                    args: launch.args.into_iter().collect(),
+                    env: launch.envs.into_iter().collect(),
+                    cwd: launch.cwd.clone(),
+                    request: "launch".to_owned(),
+                    pid: None,
+                };
+                let config = serde_json::to_string(&config).unwrap();
+
+                Ok(zed::DebugScenario {
+                    adapter: zed_scenario.adapter,
+                    label: zed_scenario.label,
+                    config,
+                    tcp_connection: None,
+                    build: None,
+                })
+            }
+            zed::DebugRequest::Attach(attach) => {
+                let config = ZigDebugConfig {
+                    program: None,
+                    pid: attach.process_id,
+                    args: Default::default(),
+                    env: Default::default(),
+                    cwd: None,
+                    request: "attach".to_owned(),
+                };
+                let config = serde_json::to_string(&config).unwrap();
+
+                Ok(zed::DebugScenario {
+                    adapter: zed_scenario.adapter,
+                    label: zed_scenario.label,
+                    config,
+                    tcp_connection: None,
+                    build: None,
+                })
+            }
+        }
+    }
+
+    fn dap_locator_create_scenario(
+        &mut self,
+        locator_name: String,
+        build_task: zed::TaskTemplate,
+        resolved_label: String,
+        debug_adapter_name: String,
+    ) -> Option<DebugScenario> {
+        if locator_name != "zig-build-run-locator" || resolved_label != "zig build run" {
+            return None;
+        }
+
+        let template = zed::BuildTaskTemplate {
+            label: "zig build".into(),
+            command: build_task.command,
+            args: vec!["build".into()],
+            env: build_task.env.clone().into_iter().collect(),
+            cwd: build_task.cwd.clone(),
+        };
+        let config = if debug_adapter_name == "CodeLLDB" {
+            serde_json::json!({
+                "sourceLanguages": ["zig"]
```

Comment:
Does CodeLLDB support zig as a source language?

Reply:
Mhm, I guess it doesn't out-of-the-box yet. CodeLLDB would need these pretty printers hooked up into its version of lldb I guess. I'll remove this option for now then.